app/javascript/spotlight/admin/croppable_modal.js

```diff
@@ -0,0 +1,104 @@
+    return regionField;
+  }
+  //When editing an existing/saved item, extract region values based on url
+  extractRegionField(iiifThumbnailUrl) {
+    if (iiifThumbnailUrl != null && iiifThumbnailUrl.length == 0) return null;
+
+    var regex = /\/[0-9]+,[0-9]+,[0-9]+,[0-9]+\//;
+    var match = iiifThumbnailUrl.match(regex);
+    return match[0].replaceAll('/', '');
+  }
+
+  getIIIFObject(dataCropperKey, itemIndex) {
+    var iiifFields = {};
+    //Retrieve the fields from the main page with the itemIndex
+    var itemElement = $('[data-cropper="' + dataCropperKey + '"]');
+    iiifFields['iiifUrlField'] = this.iiifInputField(itemIndex, 'iiif_tilesource', itemElement);
+    iiifFields['iiifManifestField'] = this.iiifInputField(itemIndex, 'iiif_manifest_url', itemElement);
+    iiifFields['iiifCanvasField'] = this.iiifInputField(itemIndex, 'iiif_canvas_id', itemElement);
+    iiifFields['iiifImageField'] = this.iiifInputField(itemIndex, 'iiif_image_id', itemElement);
+    return iiifFields;
+  }
+
+  // Field names are of the format item[item_0][iiif_image_id]
+  iiifInputField(itemIndex, fieldName, parentElement) {
+    var itemPrefix = 'item[' + itemIndex + ']';
+    var selector = 'input[name="' + itemPrefix + '[' + fieldName + ']"]';
+    return $(selector, parentElement);
+  }
+
+  attachModalSaveHandler() {
+    var context = this;
+    document.addEventListener('click', function(e) { 
+      if(e.target.matches('#blacklight-modal input#saveimage')) {
+        context.saveCroppedRegion();
+      }
+    });
+  }
+
+  saveCroppedRegion() {
+    //On hitting "save changes", we need to copy over the value
+    //to the iiif thumbnail url input field as well as the image source itself
+    var context = this;
+    var dataCropperDiv = $('#blacklight-modal [data-behavior="iiif-cropper"]');
+
+    if(dataCropperDiv) {
+      var dataCropperKey = dataCropperDiv.data("cropper-key");
+      var itemIndex = dataCropperDiv.data("index-id");
+      // Get the element on the main edit page whose select image link opened up the modal
+      var itemElement = $('[data-cropper="' + dataCropperKey + '"]');
+      // Get the hidden input field on the main edit page corresponding to this item
+      var thumbnailSaveField = context.iiifInputField(itemIndex, 'thumbnail_image_url', itemElement);
+      var fullimageSaveField = context.iiifInputField(itemIndex, 'full_image_url', itemElement);
+      var iiifTilesource = context.iiifInputField(itemIndex, 'iiif_tilesource', itemElement).val();
+      // Get the region value saved in the modal for the selected area
+      var regionElement = $('#blacklight-modal input[name="select_image_region"]');
+      var regionValue = regionElement.val();
+      // Extract the region string to incorporate into the thumbnail URL
+      var urlPrefix = iiifTilesource.substring(0, iiifTilesource.lastIndexOf('/info.json'));
+      var thumbnailUrl = urlPrefix + '/' + regionValue + '/!400,400/0/default.jpg';
```

Comment:
Should these sizes be driven by config? E.g., https://github.com/projectblacklight/spotlight/blob/40f92e657e4afdbcde9c243f1ba5bc525c57c130/lib/spotlight/engine.rb#L174

I say this with little to no understanding of how these existing work, or even if they do work...

Reply:
Okay so the Sir Trevor widgets are hardcoded to 400,400 here when they don't have a thumbnail_image_url: https://github.com/projectblacklight/spotlight/blob/9550f54007d23c05b53666ae65cf3a0690198094/app/views/spotlight/sir_trevor/blocks/_solr_documents_block.html.erb#L14

RIIIF has another hardcoded 400,400. Featured Images uses the config I've linked to above in the model. It doesn't seem like there's an easy way to resolve any of this in this issue. Unless @corylown has other ideas, I think @dnoneill is right that there's a refactor needed that is outside the scope of this. I'm happy to stick with the hardcoded values until then.